## single cell

#### Analysis of single cell RNA-seq data
https://biocellgen-public.svi.edu.au/mig_2019_scrnaseq-workshop/public/index.html

#### Single cell RNA-seq analysis notes
https://github.com/crazyhottommy/scRNAseq-analysis-notes

#### Comparing Ensembl GTF and cDNA 
https://fromsystosys.netlify.app/2020/01/31/comparing-ensembl-gtf-and-cdna/

#### Near-optimal pre-processing of single-cell RNA-seq
https://www.biorxiv.org/content/10.1101/673285v2

#### kallisto-bustools
https://bustools.github.io/BUS_notebooks_R/velocity.html
https://liorpachter.wordpress.com/2019/06/21/near-optimal-single-cell-rna-seq-pre-processing/

#### kallisto-bustools for scATAC-seq
https://f1000research.com/articles/9-199/v1


#### Benchmarking of lightweight-mapping based single-cell RNA-seq pre-processing
https://github.com/pachterlab/BP_2021

#### STARsolo: mapping, demultiplexing and quantification for single cell RNA-seq
https://github.com/alexdobin/STAR/blob/master/docs/STARsolo.md

#### cellranger vs kallisto
https://github.com/Sarah145/scRNA_pre_process

#### Evaluation of STAR and Kallisto on Single Cell RNA-Seq Data Alignment
https://twitter.com/GarmireGroup/status/1245018623783120896?s=20
https://www.g3journal.org/content/10/5/1775

#### compare kallisto-bustools and cellranger for single nuclei sequencing data
https://divingintogeneticsandgenomics.rbind.io/post/compare-kallisto-bustools-and-cellranger-for-single-nuclei-seqencing-data/

#### differences between gene counts when using STARsolo Gene and GeneFull options
https://github.com/tzuni/starsolo_quant_analysis/blob/master/notebooks/star_solo_gene_genefull_plot.ipynb

##### Aligner Comparsion project code summary
https://github.com/yhdu36/aligner

#### Preprocessing choices affect RNA velocity results for droplet scRNA-seq data
https://github.com/csoneson/rna_velocity_quant

#### Single-cell RNA-seq for dummies
https://liorpachter.wordpress.com/2019/06/21/single-cell-rna-seq-for-dummies/

#### How to solve an NP-complete problem in linear time
https://liorpachter.wordpress.com/2019/06/21/how-to-solve-an-np-complete-problem-in-linear-time/

#### Rotating the knee (plot) and related yoga
https://liorpachter.wordpress.com/2019/06/24/rotating-the-knee-plot-and-related-yoga/

#### High velocity RNA velocity
https://liorpachter.wordpress.com/2019/07/01/high-velocity-rna-velocity/

#### Is Python really faster than C++ ?
https://liorpachter.wordpress.com/2019/10/09/is-python-really-faster-than-c/


#### tidyseurat provides a bridge between the Seurat single-cell package [@butler2018integrating; @stuart2019comprehensive] and the tidyverse 
https://cran.r-project.org/web/packages/tidyseurat/vignettes/introduction.html

#### Cell hashing 
https://btep.ccr.cancer.gov/wp-content/uploads/cihan_cell_hashing_slides_oct10_fnl.pdf
#### Rotating the knee (plot) and related yoga
https://liorpachter.wordpress.com/2019/06/24/rotating-the-knee-plot-and-related-yoga/amp/?__twitter_impression=true

#### Deatled codes and data in paper Benchmarking Computational Doublet-Detection Methods for Single-Cell RNA Sequencing Data
https://arxiv.org/pdf/2101.08860.pdf

#### TooManyCells identifies and visualizes relationships of single-cell clades
https://www.nature.com/articles/s41592-020-0748-5
https://github.com/faryabib/too-many-cells

#### VeloViz creates RNA-velocity-informed 2D embeddings for single cell transcriptomics data.
https://jef.works/veloviz/
https://jef.works/veloviz/pancreas
https://jef.works/veloviz/merfish
https://jef.works/veloviz/simulation

#### HoneyBADGER HoneyBADGER (hidden Markov model integrated Bayesian approach for detecting CNV and LOH events from single-cell RNA-seq data) identifies and infers the presence of CNV and LOH events in single cells and reconstructs subclonal architecture using allele and expression information from single-cell RNA-sequencing data.
https://jef.works/HoneyBADGER/

#### scDataviz: single cell dataviz and downstream analyses
https://bioconductor.org/packages/release/bioc/vignettes/scDataviz/inst/doc/scDataviz.html

#### All code, processed data and underlying structures of "Mapping the Immune Environment in Clear Cell Renal Carcinoma by Single-Cell Genomics"
https://zenodo.org/record/4311825
#### Collections of Benchmark studies 
https://www.biomedcentral.com/collections/benchmarkingstudies

#### 12 Grand Challenges in Single-Cell DataScience
https://peerj.com/preprints/27885.pdf

#### Machine learning single cell courses
https://www.krishnaswamylab.org/workshop/2019

#### Use a Negative Binomial for Count Data
https://towardsdatascience.com/use-a-negative-binomial-for-count-data-c68c062de203

#### Scater: pre-processing, quality control, normalisation and visualisation of single-cell RNA-seq data in R
http://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html

#### Gene Discovery Rate in RNA-seq data
http://www.nxn.se/valent/gene-discovery-rate-in-rna-seq-data

#### The first steps in RNA-seq expression analysis (single-cell and other)
http://www.nxn.se/valent/2016/10/3/the-first-steps-in-rna-seq-expression-analysis-single-cell-and-other

## single cell & bayesian
#### a blog must read blog
https://alanocallaghan.github.io/
### BASiCS (Bayesian Analysis of Single-Cell Sequencing data)
https://bioconductor.org/packages/release/bioc/html/BASiCS.html
http://bioconductor.org/packages/release/bioc/vignettes/BASiCS/inst/doc/BASiCS.html
